packages/library/src/commonMain/kotlin/io/realm/internal/SuspendableNotifier.kt

```diff
@@ -240,19 +240,30 @@ internal class SuspendableNotifier(
      * FIXME Callers of this method must make sure it is called on the correct [SuspendableNotifier.dispatcher].
      */
     internal fun <T> registerListChangedListener(
-        list: RealmList<T>,
+        listDelegate: ManagedRealmList<T>,
         callback: Callback<RealmList<T>?>
     ): Cancellable {
-        val liveList = list.thaw(realm.realmReference)
+        val liveListPointer = RealmInterop.realm_list_thaw(
+            listDelegate.nativePointer,
+            realm.realmReference.dbPointer
+        )
+        val liveListDelegate = ManagedRealmList<T>(
+            liveListPointer,
+            listDelegate.metadata.copy(realm = realm.realmReference)
+        )
         return registerChangedListener(
-            liveComponentPointer = liveList.listPtr,
+            liveComponentPointer = liveListPointer,
             notifyComponentUpdate = { frozenRealm ->
-                if (!liveList.isValid()) {
+                if (!liveListDelegate.isValid()) {
                     callback.onChange(null)
                 } else {
                     // Notifications need to be delivered with the version they where created on, otherwise
                     // the fine-grained notification data might be out of sync.
-                    val frozenList = liveList.freeze(frozenRealm)
+                    val frozenListPointer = RealmInterop.realm_list_freeze(
```

Comment:
Shouldn't this logic be inside the `liveListDelegate.freeze` method?

Reply:
Yep. Just migrated a previous sample implementation, so didn't gave it too much thought.